bool forward
bool forward_left
bool forward_right
bool backward